zhaquirks/tuya/ts0601_trv_sas.py

```diff
@@ -49,63 +37,62 @@
 
 
 class ManufacturerThermostatCluster(TuyaManufClusterAttributes):
-    """Manufacturer thermostat cluster."""
+    """Tuya manufacturer specific cluster."""
+
+    class State(t.enum8):
+        """State option."""
+
+        Off = 0x00
+        On = 0x01
+
+    class BatteryState(t.enum8):
+        """Battery state option."""
+
+        Normal = 0x00
+        Low = 0x01
+
+    class ScheduleState(t.enum8):
+        """Schedule state option."""
+
+        Disabled = 0x00
+        Enabled = 0x01
 
     attributes = TuyaManufClusterAttributes.attributes.copy()
     attributes.update(
         {
-            SYSTEM_MODE_COMMAND_ID: ("system_mode", t.uint8_t, True),
-            OCCUPIED_HEATING_SETPOINT_COMMAND_ID: (
-                "occupied_heating_setpoint",
-                t.uint32_t,
-                True,
-            ),
-            LOCAL_TEMP_COMMAND_ID: ("local_temperature", t.uint32_t, True),
-            BATTERY_STATE_COMMAND_ID: ("battery_state", t.uint8_t, True),
+            SASSWEL_STATE_ATTR: ("state", State, True),
+            SASSWEL_HEATING_SETPOINT_ATTR: ("heating_setpoint", t.uint32_t, True),
```

Comment:
I see how that's tricky, let me try to explain.

ZCL Thermostat class uses `occupied_heating_setpoint` this is the attribute name which zigpy sends to the quirk when user requests setpoint change. Unit used by ZCL is centidegree Celsius (22,4 °C = 2240)

Tuya thermostat uses different set of attributes and the name used by Tuya developers is simply "heating setpoint", I've used different name to avoid confusion between ZCL and Tuya attributes which use different units. Unit used by Tuya is decidegree Celsius (22,4 °C = 224).

When zigpy updates the value for ZCL Thermostat, (i.e. when user changes setpoint in HA dashboard) it calls `ThermostatCluster.map_attribute()` with `attribute = "occupied_heating_setpoint"`. This method converts the value from centidegree to decidegree and translates ZCL's `occupied_heating_setpoint` to Tuya's `SASSWEL_HEATING_SETPOINT_ATTR`.

On the other hand device reports two temperature values with attrids: `SASSWEL_HEATING_SETPOINT_ATTR` and `SASSWEL_LOCAL_TEMP_ATTR`. First is the current heating setpoint and the second is the local temperature as measured by the temperature sensor on the device. Both reported values are handled by the same event listener implemented in `TuyaThermostatCluster.temperature_change()`. To make code simpler both reports are handled by lines 78:83 and Tuya->ZCL attribute name translation is handled by the `TEMPERATURE_ATTRS` mapping, both values require the same *10 conversion from Tuya to ZCL standard.